frida/src/lib.rs

```diff
@@ -96,8 +120,59 @@ impl Frida {
     }
 }
 
-impl Drop for Frida {
+// A marker type that exists only while Gum is initialized.
+struct FridaSingleton;
+
+impl FridaSingleton {
+    fn new() -> Self {
+        unsafe { frida_sys::frida_init() };
+        FridaSingleton
+    }
+}
+
+impl Drop for FridaSingleton {
     fn drop(&mut self) {
-        unsafe { frida_sys::frida_deinit() };
+        unsafe { frida_sys::frida_deinit() }
+    }
+}
+
+static THE_ONE_TRUE_FRIDA: Mutex<Option<Arc<FridaSingleton>>> = Mutex::new(None);
```

Comment:
While this name is cute and pithy, would be better to call it `FRIDA_SINGLETON`, I think.